spec/mongoid/fields/field_types_spec.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Mongoid::Fields::FieldTypes do
+
+  after do
+    described_class.instance_variable_set(:@mapping, described_class::DEFAULT_MAPPING.dup)
+  end
+
+  describe '.get' do
+    subject { described_class.get(type) }
+
+    context 'when value is a default mapped symbol' do
+      let(:type) { :float }
+
+      it 'uses the default mapped type' do
+        is_expected.to eq Float
+      end
+    end
+
+    context 'when value is a default mapped string' do
+      let(:type) { 'double' }
+
+      it 'uses the default mapped type' do
+        is_expected.to eq Float
+      end
+    end
+
+    context 'when value is a custom mapped symbol' do
+      before { described_class.define('number', Integer) }
+      let(:type) { :number }
+
+      it 'uses the custom mapped type' do
+        is_expected.to eq Integer
+      end
+    end
+
+    context 'when value is a custom mapped string' do
+      before { described_class.define(:number, Float) }
+      let(:type) { 'number' }
+
+      it 'uses the custom mapped type' do
+        is_expected.to eq Float
+      end
+    end
+
+    context 'when value is an unmapped symbol' do
+      let(:type) { :my_value }
+
+      it 'returns nil' do
+        is_expected.to eq nil
+      end
+    end
+
+    context 'when value is a unmapped string' do
+      let(:type) { 'my_value' }
+
+      it 'returns nil' do
+        is_expected.to eq nil
+      end
+    end
+
+    context 'when value is a module' do
+      let(:type) { String }
+
+      it 'uses the module type' do
+        is_expected.to eq String
+      end
+    end
+
+    context 'when value is the module Boolean' do
+      let(:type) do
+        stub_const('Boolean', Module.new)
```

Comment:
What is the purpose of this stub?

Reply:
The constant Boolean does not exist in the global scope. If I comment out this line:

```
  0) Mongoid::Fields::FieldTypes.get when value is the module Boolean returns Mongoid::Boolean type
     Failure/Error: Boolean

     NameError:
       uninitialized constant Boolean
     # ./spec/mongoid/fields/field_types_spec.rb:78:in `block (4 levels) in <top (required)>'
```

Reply:
Please mark resolved.